pkg/driver/node/mounter/pod_mounter.go

```diff
@@ -0,0 +1,403 @@
+	// there is an existing mount point at `target`.
+	credEnv, authenticationSource, err := pm.credProvider.Provide(ctx, credentialCtx)
+	if err != nil {
+		klog.V(4).Infof("NodePublishVolume: Failed to provide credentials for %s: %v", target, err)
+		return fmt.Errorf("Failed to provide credentials for %q: %w", target, err)
+	}
+
+	if isMountPoint {
+		klog.V(4).Infof("Target path %q is already mounted", target)
+		return nil
+	}
+
+	env := envprovider.Default()
+	env.Merge(credEnv)
+
+	// Move `--aws-max-attempts` to env if provided
+	if maxAttempts, ok := args.Remove(mountpoint.ArgAWSMaxAttempts); ok {
+		env.Set(envprovider.EnvMaxAttempts, maxAttempts)
+	}
+
+	args.Set(mountpoint.ArgUserAgentPrefix, UserAgent(authenticationSource, pm.kubernetesVersion))
+
+	podMountSockPath := mppod.PathOnHost(podPath, mppod.KnownPathMountSock)
+	podMountErrorPath := mppod.PathOnHost(podPath, mppod.KnownPathMountError)
+
+	klog.V(4).Infof("Waiting for Mountpoint Pod %s to be ready to accept connections on %s", pod.Name, podMountSockPath)
+
+	// Wait until Mountpoint Pod is ready to accept connections
+	err = util.WaitForUnixSocket(mountpointPodReadinessTimeout, mountpointPodReadinessCheckInterval, podMountSockPath)
+	if err != nil {
+		return fmt.Errorf("Failed to wait for Mountpoint Pod to be ready in given timeout for %q: %w", target, err)
+	}
+
+	klog.V(4).Infof("Mounting %s for %s", target, pod.Name)
+
+	fuseDeviceFD, err := pm.mountSyscallWithDefault(target, args)
+	if err != nil {
+		if fuseDeviceFD > 0 {
+			pm.closeFUSEDevFD(fuseDeviceFD)
+		}
+		return fmt.Errorf("Failed to mount %s: %w", target, err)
+	}
+
+	// This will set to false in the success condition. This is set to `true` by default to
+	// ensure we don't leave `target` mounted if Mountpoint is not started to serve requests for it.
+	unmount := true
+	defer func() {
+		if unmount {
+			if err := pm.unmountTarget(target); err != nil {
+				klog.V(4).ErrorS(err, "Failed to unmount mounted target %s\n", target)
+			} else {
+				klog.V(4).Infof("Target %s unmounted successfully\n", target)
+			}
+		}
+	}()
+
+	// This function can either fail or successfully send mount options to Mountpoint Pod - in which
+	// Mountpoint Pod will get its own fd referencing the same underlying file description.
+	// In both case we need to close the fd in this process.
+	defer pm.closeFUSEDevFD(fuseDeviceFD)
+
+	// Remove old mount error file if exists
+	_ = os.Remove(podMountErrorPath)
+
+	klog.V(4).Infof("Sending mount options to Mountpoint Pod %s on %s", pod.Name, podMountSockPath)
+
+	err = mountoptions.Send(ctx, podMountSockPath, mountoptions.Options{
+		Fd:         fuseDeviceFD,
+		BucketName: bucketName,
+		Args:       args.SortedList(),
+		Env:        env.List(),
+	})
+	if err != nil {
+		klog.V(4).Infof("Failed to send mount option to Mountpoint Pod %s for %s: %s\n", pod.Name, target, err)
+		return fmt.Errorf("Failed to send mount options to Mountpoint Pod %s for %s: %w", pod.Name, target, err)
```

Comment:
Why does this one use %w at the end and the info use %s?

Reply:
`%w` is for [wrapping the underlying error](https://pkg.go.dev/fmt#Errorf) to be able to unwrap later via [`errors` package](https://pkg.go.dev/errors). The klog package doesn't support it and just asks for `%s` or `%v`.